library/network_connections.py

```diff
@@ -966,12 +970,15 @@ def connection_create(self, connections, idx, connection_current=None):
             for s in ip["dns_search"]:
                 s_ip4.add_dns_search(s)
 
-            if ip["auto6"]:
+            if ip["ipv6_disabled"]:
+                s_ip6.set_property(NM.SETTING_IP_CONFIG_METHOD, "disabled")
+            elif ip["auto6"]:
                 s_ip6.set_property(NM.SETTING_IP_CONFIG_METHOD, "auto")
             elif addrs6:
                 s_ip6.set_property(NM.SETTING_IP_CONFIG_METHOD, "manual")
             else:
                 s_ip6.set_property(NM.SETTING_IP_CONFIG_METHOD, "ignore")
```

Comment:
in a follow-up patch, it should be considered that "ignore" gets dropped and instead have "link-local" here. Possibly "ignore" setting should not be configurable via the role. That would be a change in behavior and warrants a separate commit. But maybe you can add a `# TODO` comment now.

Reply:
done.